### PR TITLE
feat(fa): fconfirmation page

### PR DESCRIPTION
**Pull request by davidpxn**
# ... 👆

https://app.asana.com/0/1202024607727436/1201706500425579

## What

Display correct messages and alerts on the confirmation page.

## Why

So users get relevant information based on their roles.

## Screenshots / Gifs

Applicant success
image

Applicant missing income files
image

Applicant (with spouse) success
image

Applicant (with spouse) missing files
image

Spouse success
image

Spouse missing files
image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    answers.income === ApproveOptions.Yes && !hasFiles('incomeFiles', answers)
+
+  let firstStepText
+  if (applicantHasSpouse && missingIncomeFiles) {
+    firstStepText = confirmation.nextSteps.contentBothMissingFiles
+  } else if (applicantHasSpouse) {
+    firstStepText = confirmation.nextSteps.contentSpouseMissingFiles
+  } else if (missingIncomeFiles) {
+    firstStepText = confirmation.nextSteps.contentMissingFiles
+  }
```
Comment (MargretFinnboga) [inline libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx:22]: should this be a switch statement?

File: libs/application/templates/financial-aid/src/fields/Confirmation/Confirmation.tsx [hunk 2]
```diff
@@ -1,40 +1,77 @@
...
-import { useIntl } from 'react-intl'
+
 import { confirmation, copyUrl } from '../../lib/messages'
 import { DescriptionText, ConfirmationSectionImage, CopyUrl } from '..'
-import { hasSpouse } from '../../lib/utils'
 
-const Confirmation = ({ application }: FAFieldBaseProps) => {
+interface Props {
+  firstStepText?: MessageDescriptor
+  missingIncomeFiles: boolean
```
Comment (MargretFinnboga) [inline libs/application/templates/financial-aid/src/fields/Confirmation/Confirmation.tsx:12]: should we maybe hasIncomeFiles keep the naming consistant, so hasIncomeFiles?

**Review by MargretFinnboga — COMMENTED** [review]
Two questions 🙋🏼

File: libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx (hunk 1 above)
Comment (davidpxn) [reply libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx:22]: how do you use a switch statement with two variables? are you talking about like this:
image

Comment (MargretFinnboga) [reply libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx:22]: yesss exactly like this ☝🏼

File: libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    case (applicantHasSpouse && missingIncomeFiles):
+      firstStepText = confirmation.nextSteps.contentBothMissingFiles
+      break
+    case (applicantHasSpouse):
+      firstStepText = confirmation.nextSteps.contentSpouseMissingFiles
+      break
+    case (missingIncomeFiles):
+      firstStepText = confirmation.nextSteps.contentMissingFiles
+      break
+  }
```
Comment (MargretFinnboga) [inline libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx:26]: const firstStepText =  switch (true) {
    case (applicantHasSpouse && missingIncomeFiles):
      return confirmation.nextSteps.contentBothMissingFiles
     
Was more what I was thinking 🤔

Comment (davidpxn) [reply libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx:26]: I don't think that syntax is allowed. The closest thing to this that I can come up with is:
image

Comment (MargretFinnboga) [reply libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx:26]: yes it was just a pseudo code as an example, yea i would take out the break and have return and have not have firstStepText as string rather, I suggest having it as a function that return a string like in the example you made above 🙌🏼

**Review by MargretFinnboga — COMMENTED** [review]
just minor comment 🤠

File: libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  const firstStepText = () => {
+    switch (true) {
+      case (applicantHasSpouse && missingIncomeFiles):
+        return confirmation.nextSteps.contentBothMissingFiles
+      case (applicantHasSpouse):
+        return confirmation.nextSteps.contentSpouseMissingFiles
+      case (missingIncomeFiles):
+        return confirmation.nextSteps.contentMissingFiles
+      default:
+        return undefined
```
Comment (MargretFinnboga) [inline libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx:24]: is this necessary? if you take it doesn't it become undefined?

**Review by MargretFinnboga — APPROVED** [review]
One more minor comment, just a wondering 👩🏼‍🚀

File: libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx (hunk 4 above)
Comment (davidpxn) [reply libs/application/templates/financial-aid/src/fields/Confirmation/ApplicantConfirmation.tsx]: It was giving me an error if I didn't have the default but I just tried to remove it and now it accepts it. Don't know what was happening 🤷‍♂️

**Pull request merged**